src/scry/implementations.cr

```diff
@@ -49,15 +69,19 @@ module Scry
     private def analyze(filename, position, scope)
       response = crystal_tool(filename, position, scope)
       Log.logger.debug("response: #{response}")
-      parsed_response = JSON.parse(response)
-      implementations = parsed_response["implementations"]?
-      if implementations
-        implementations
-      else
+      response = (Array(BuildFailure) | ImplementationsResponse).from_json(response)
+      case response
+      when .is_a?(Array(BuildFailure))
```

Comment:
I think you can use `when Array(BuildFailure)` directly

Reply:
You're right, Done! :+1: